includes/class-wc-payments-captured-event-note.php

```diff
@@ -173,15 +173,33 @@ public function compose_fee_break_down() {
 	public function compose_net_string(): string {
 		$data = $this->captured_event['transaction_details'];
 
-		$net = WC_Payments_Utils::interpret_stripe_amount( (int) $data['store_amount'] - $data['store_fee'], $data['store_currency'] );
+		// Determine the type of payment and select the appropriate amounts and currencies.
+		if ( $this->is_fx_event() ) {
+			// For fx events, we need the store amount and currency to display the net amount
+			// in the store currency.
+			$amount_key          = 'store_amount';
```

Comment:
It is the last nudge about complexity (and how friendly we are with Static code analysis). We'd better refind the approach in this if-else statement:

```php
    $amount_key = 'store_amount'; // before

    $amount = $data[ 'store_amount' ]; // after
```

Reply:
Good point, followed your suggestion here https://github.com/Automattic/woocommerce-payments/pull/8107/commits/b7433f38d7648142a233a20a32c75c731a69dca0 thanks!